### PR TITLE
Ashishonce/modelprofile

### PR DESCRIPTION
**Pull request by ashishonce**
adding support for profiling model

### CODE REVIEW

File: code/main.py [hunk 1]
```diff
@@ -177,6 +177,24 @@ def main():
...
+            if dataset is not None:
+                profile_datasets.append((f"{dataset_name}", dataset))
+        default_profileName = model_name + "_profile"
+        profileName = parameters.get("profileName", default_profileName)
+        profile = Model.profile(ws, profileName, [model], inference_config, input_dataset=profile_datasets)
+        if parameters.get("wait_for_profiling_completion", False):
+            profile.wait_for_profiling(True)
+            profiling_details = profile.get_details()
+            print(profiling_details)
+
```
Comment (marvinbuss) [inline code/main.py:197]: Let me work on some additions here later this week.

File: .cloud/.azure/aks_deploy.json [hunk 2]
```diff
@@ -19,6 +19,10 @@
     "memory_gb": 0.5,
     "gpu_cores": 0,
     "delete_service_after_test": false,
+    "profiling_enabled": false,
+    "profile_name": "<your-profiling-name>",
+    "profile_dataset" : "[<data set to use with profiling>]",
```
Comment (marvinbuss) [inline .cloud/.azure/aks_deploy.json:24]: Should this be a list or a single dataset name?

**Comment by ashishonce** [comment]
> Please review comments.
> Also:
> 
> 1. Documentation is fully missing
> 2. Schema needs update.

Yes I have just raised it to give a idea, 
also the code needupdate as we are not setting up other fields , 
I think as number of fields are more for profiling as well, shall we consider moving to a different action or at least another file like profile.json? 

currently I am not using other fields to Model.profile(..) function .

**Comment by ashishonce** [comment]
there is one more thing in this. profiling can be done separately and we are making people provide deploy information for just profiling , if he just want to profile. 
so making another action is a better way to do it.

**Comment by ashishonce** [comment]
> there is one more thing in this. profiling can be done separately and we are making people provide deploy information for just profiling , if he just want to profile.
> so making another action is a better way to do it.

Will deploy inference config wont be different than profiling inference config. as I imagine profiling to run as a separate workflow , with lot more parallel inputs

**Comment by marvinbuss** [comment]
> > Please review comments.
> > Also:
> > 
> > 1. Documentation is fully missing
> > 2. Schema needs update.
> 
> Yes I have just raised it to give a idea,
> also the code needupdate as we are not setting up other fields ,
> I think as number of fields are more for profiling as well, shall we consider moving to a different action or at least another file like profile.json?
> 
> currently I am not using other fields to Model.profile(..) function .
> 
> Also want to check if schema validation is actually working.
> as for all our test one value **"delete_service_after_deployment"** is different than what was provided in the test "**delete_service_after_test"** and it still passed?

I created another PR to update the parameter names in the files.
Let's include the functionality here for now. The feature has a lot of limitations and I don't think we should give it that much attention yet. Once there is a need, we can create a separate Action for the feature.

**Comment by marvinbuss** [comment]
@ashishonce Can you please review quickly?

**Pull request merged**